helix-term/src/application.rs

```diff
@@ -446,7 +452,7 @@ impl Application {
     #[cfg(not(windows))]
     pub async fn handle_signals(&mut self, signal: i32) {
         match signal {
-            signal::SIGTSTP => {
+            signal::SIGTSTP | signal::SIGTERM | signal::SIGINT => {
```

Comment:
This block handles suspending the process (with `C-z`) rather than terminating it. SIGTERM and SIGINT should be handled in a new block that calls `self.restore_term().unwrap()` and then exits the event loop. That will also need some changes in `event_loop_until_idle`